tree/dataframe/src/RDFUtils.cxx

```diff
@@ -137,7 +137,8 @@ std::string ComposeRVecTypeName(const std::string &valueType)
 
 std::string GetLeafTypeName(TLeaf *leaf, const std::string &colName)
 {
-   std::string colType = leaf->GetTypeName();
+   const char *colTypeCStr = leaf->GetTypeName();
+   std::string colType = colTypeCStr == nullptr ? "" : colTypeCStr;
    if (colType.empty())
       throw std::runtime_error("Could not deduce type of leaf " + colName);
```

Comment:
What about:
```suggestion
   const char *colTypeCStr = leaf->GetTypeName();
   if (!colTypeCStr)
      throw std::runtime_error("Could not deduce type of leaf " + colName);
   std::string colType = colTypeCStr;
```

Reply:
I never realized that with `const char*`s you can check for empty string and null at the same time :D

Reply:
Oh you can't - then it's `if (!colTypeCStr || !colTypeCStr[0])`

Reply:
...and then the difference to your version isn't that noticeable anymore.

Reply:
...but it looks like you can?

```
~ root -l
root [0] const char* foo = ""
(const char *) ""
root [1] bool(foo)
(bool) false
root [2]
```

Reply:
ah no that...doesn't work as I thought